target_chains/ethereum/contracts/contracts/pyth/PythAccumulator.sol

```diff
@@ -105,29 +125,56 @@ abstract contract PythAccumulator is PythGetters, PythSetters, AbstractPyth {
             if (accumulatorUpdate.length < offset)
                 revert PythErrors.InvalidUpdateData();
 
-            if (updateType == UpdateType.WormholeMerkle) {
-                updatePricesUsingWormholeMerkle(
-                    UnsafeBytesLib.slice(
-                        accumulatorUpdate,
-                        offset,
-                        accumulatorUpdate.length - offset
-                    )
-                );
-            } else {
+            if (updateType != UpdateType.WormholeMerkle) {
```

Comment:
This check seems like it's in the wrong place. If the updateType is a different type of payload, it's still a valid header. I think this type needs to be returned so you can do the check on line 56 above